pkg/steps/template_test.go

```diff
@@ -0,0 +1,173 @@
+package steps
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/openshift/ci-operator/pkg/api"
+	"k8s.io/apimachinery/pkg/util/diff"
+)
+
+func TestDeferredParametersAllLinks(t *testing.T) {
+	var testCases = []struct {
+		purpose       string
+		dp            *DeferredParameters
+		expectedItems int
+	}{{
+		purpose: "sanity",
+		dp: &DeferredParameters{
+			links: map[string][]api.StepLink{
+				"K1": {someStepLink("ONE"), someStepLink("TWO")},
+				"K2": {someStepLink("THREE")},
+			},
+		},
+		// only compare count of returned items because AllLinks can
```

Comment:
Can sort for deterministic ordering

Reply:
Not that simply, seems like I would need to implement some interface to make those classes sortable and I was lazy. I'll consider it (and maybe I'll even get rid of misusing the "real" StepLink structs and implement a simple one for testing).

Reply:
IMO not worth the effort, `StepLink` is an interface that can do only `Same` and `Matches`, there's nothing to sort them by easily :/